codesign/lib/src/transfer_utils.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:process/process.dart';
+
+import './utils.dart';
+
+/// Utility function class to handle upload/download of files from/to google cloud.
+class EngineArtifactTransfer {
+  EngineArtifactTransfer({
+    required this.gsCloudBaseUrl,
```

Comment:
You could simplify this by only passing the gcs bucket.

Then, internally you'd need to rewrite some of the urls to be `gs://$bucket/$path`

Reply:
Sure, simplified constructor parameters.